### PR TITLE
Upgrade Aeron and make use of channelStatus on subscription

### PR DESCRIPTION
**Pull request by chbatey**
Aeron 1.9.1 is wire compatible with 1.7 so nothing to do when doing an rolling upgrade.

We asked for an easier way to get hold of a channel status. It was added to subscription which isn't that easy for us to get hold of but used it anyway rather than scanning all counters.

### CODE REVIEW

File: akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala [hunk 1]
```diff
@@ -302,6 +301,8 @@ private[remote] abstract class ArteryTransport(_system: ExtendedActorSystem, _pr
   import ArteryTransport._
   import FlightRecorderEvents._
 
+  type LifeCycle
```
Comment (ktoso) [inline akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala]: internal api?

Comment (chbatey) [reply akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala]: The whole type is Internal API and private to remote

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-remote/src/main/scala/akka/remote/artery/aeron/ArteryAeronUdpTransport.scala [hunk 2]
```diff
@@ -176,7 +177,7 @@ private[remote] class ArteryAeronUdpTransport(_system: ExtendedActorSystem, _pro
         // freeSessionBuffer in AeronSource FragmentAssembler
         streamMatValues.get.valuesIterator.foreach {
           case InboundStreamMatValues(resourceLife, _) ⇒
-            resourceLife.foreach(_.onUnavailableImage(img.sessionId))
+            resourceLife.onUnavailableImage(img.sessionId)
```
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/artery/aeron/ArteryAeronUdpTransport.scala]: nice

File: akka-remote/src/main/scala/akka/remote/artery/aeron/ArteryAeronUdpTransport.scala [hunk 3]
```diff
@@ -219,13 +220,14 @@ private[remote] class ArteryAeronUdpTransport(_system: ExtendedActorSystem, _pro
...
-    val counterIdForInboundChannel = findCounterId(s"rcv-channel: $inboundChannel")
+    val aeronLifecyle = streamMatValues.get()(ControlStreamId).lifeCycle
+
     val waitInterval = 200
     val retries = math.max(1, settings.Bind.BindTimeout.toMillis / waitInterval)
     retry(retries)
 
     @tailrec def retry(retries: Long): Unit = {
-      val status = aeron.countersReader().getCounterValue(counterIdForInboundChannel)
+      val status = Await.result(aeronLifecyle.channelEndpointStatus(), settings.Bind.BindTimeout)
```
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/artery/aeron/ArteryAeronUdpTransport.scala]: ok, this is thread that we may block anyway in the startup

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
Create issue https://github.com/akka/akka/issues/25059

**Comment by patriknw** [comment]
Refs https://github.com/akka/akka/issues/25059

**Pull request merged**